app/controllers/sign_up_controller.rb

```diff
@@ -41,4 +37,11 @@ def confirm_email
   def user_params
     params.permit(:email, :name, :password, :gender, :birthday, :phone)
   end
+
+  def encode_jwt_token
```

Comment:
It's better to move this method somewhere in ApplicationController or module to reuse this. As a parameter method can receive user. Also refactor the other usages of such method.

Reply:
Updated: Method moved to ApplicationController.  Added parameter for user recieve. This method has been used  for login only. Login action updated.